src/main/java/org/folio/innreach/domain/service/impl/MARCTransformationOptionsSettingsServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+    marcTransformOptSet.setCentralServer(centralServerRef(centralServerId));
+    var createdMARCTransformOptSet = repository.save(marcTransformOptSet);
+    return mapper.toDto(createdMARCTransformOptSet);
+  }
+
+  @Override
+  public MARCTransformationOptionsSettingsDTO update(UUID centralServerId, MARCTransformationOptionsSettingsDTO marcTransformOptSetDTO) {
+    var marcTransformOptSet = repository.findOneByCentralServerId(centralServerId).orElseThrow(
+      () -> new EntityNotFoundException(TEXT_MARC_TRANSFORM_OPT_SET_NOT_FOUND + centralServerId));
+
+    var updated = mapper.toEntity(marcTransformOptSetDTO);
+    updateMARCTransformOptSet(marcTransformOptSet, updated);
+    updateModifiedFields(marcTransformOptSet, updated);
+
+    repository.save(marcTransformOptSet);
+
+    return mapper.toDto(marcTransformOptSet);
+  }
+
+  @Override
+  @Transactional
+  public void delete(UUID centralServerId) {
+    var marcTransformOptSet = repository.findOneByCentralServerId(centralServerId).orElseThrow(
+      () -> new EntityNotFoundException(TEXT_MARC_TRANSFORM_OPT_SET_NOT_FOUND + centralServerId));
+    repository.delete(marcTransformOptSet);
+  }
+
+  @Override
+  public MARCTransformationOptionsSettingsListDTO getAll(int offset, int limit) {
+    var marcTransformOptSetList = collectMARCTransformOptSet(offset, limit);
+
+    var marcTransformOptSetListDTO = new MARCTransformationOptionsSettingsListDTO();
+    marcTransformOptSetListDTO.setMaRCTransformOptSetList(marcTransformOptSetList);
+    marcTransformOptSetListDTO.setTotalRecords(marcTransformOptSetList.size());
+
+    return marcTransformOptSetListDTO;
+  }
+
+  private List<MARCTransformationOptionsSettingsDTO> collectMARCTransformOptSet(Integer offset, Integer limit) {
+    return repository.findAll(PageRequest.of(offset, limit))
+      .stream()
+      .map(mapper::toDto)
+      .collect(Collectors.toList());
+  }
+
+  private void updateMARCTransformOptSet(MARCTransformationOptionsSettings marcTransformOptSet, MARCTransformationOptionsSettings updated) {
+    marcTransformOptSet.setExcludedMARCFields(updated.getExcludedMARCFields());
+    marcTransformOptSet.setConfigIsActive(updated.getConfigIsActive());
+  }
+
+  private void updateModifiedFields(MARCTransformationOptionsSettings marcTransformOptSet, MARCTransformationOptionsSettings updated) {
+    var currentModifiedFields = new ArrayList<>(marcTransformOptSet.getModifiedFieldsForContributedRecords());
+    var updatedModifiedFields = updated.getModifiedFieldsForContributedRecords();
+
+    ServiceUtils.merge(updatedModifiedFields, currentModifiedFields, comparator, marcTransformOptSet::addModifiedFieldForContributedRecords,
+      this::updateModifiedFields1, marcTransformOptSet::removeModifiedFieldForContributedRecords);
+  }
+
+  private void updateModifiedFields1(FieldConfiguration incoming, FieldConfiguration existing) {
```

Comment:
it's better to avoid distinguishing between methods using numeric suffix. 
Lets rename this to `mergeModifiedFields`